internal/golang/encoding/json/kubernetes_patch.go

```diff
@@ -118,10 +124,29 @@ func (d *decodeState) appendStrictFieldStackIndex(i int) {
 	d.strictFieldStack = append(d.strictFieldStack, "[", strconv.Itoa(i), "]")
 }
 
+type strictErrType string
+
+const (
+	unknownStrictErrType   strictErrType = "unknown field"
+	duplicateStrictErrType               = "duplicate field"
+)
+
+// StrictError is a strict decoding error
+// It has an ErrType (either unknown or duplicate)
+// and a path to the erroneous field
+type StrictError struct {
+	ErrType strictErrType
+	Path    string
+}
+
+func (e *StrictError) Error() string {
+	return string(e.ErrType) + ` "` + e.Path + `"`
```

Comment:
quote to keep current behavior:

```suggestion
	return string(e.ErrType) + ` "` + strconv.Quote(e.Path) + `"`
```